### PR TITLE
Optimize the hell out of SSH calls

### PR DESCRIPTION
**Pull request by thieman**
@paetling Running `dusty test insights all`, which runs an empty test module, takes 11.7 seconds before my changes and 4.6 seconds after them.

### CODE REVIEW

File: dusty/systems/virtualbox/__init__.py [hunk 1]
```diff
@@ -152,12 +166,30 @@ def initialize_docker_vm():
 
 @memoized
 def get_docker_vm_ip():
+    ssh_port = None
+    for line in _get_vm_config():
+        if line.startswith('Forwarding'):
```
Comment (paetling) [inline dusty/systems/virtualbox/__init__.py]: what is this now?

Comment (thieman) [inline dusty/systems/virtualbox/__init__.py]: It's a line in the massive dump that VBox gives you for the machine config. We're taking advantage of the fact that Machine (or maybe just VBox) rigs up a localhost forwarding rule for SSH to the target machine. We use this to figure out WTF the target machine's actual IP is, which we need if we want to talk to it on any port other than the one that gets forwarded for us.

File: dusty/systems/virtualbox/__init__.py [hunk 2]
```diff
@@ -152,12 +166,30 @@ def initialize_docker_vm():
...
+                break
+
+    if ssh_port:
+        ip_addr = check_output_demoted(['ssh', '-i', _vm_key_path(), '-p', ssh_port,
+                                        'docker@127.0.0.1', 'ip addr show dev eth1'])
+        for line in ip_addr.splitlines():
+            line = line.strip()
+            if line.startswith('inet') and not line.startswith('inet6'):
+                ip = line.split(' ')[1].split('/')[0]
+                return ip
```
Comment (paetling) [inline dusty/systems/virtualbox/__init__.py]: add some comments on why these work?

Comment (thieman) [inline dusty/systems/virtualbox/__init__.py]: Oh was this not clear?

**Pull request merged**